### PR TITLE
Migrate Extending Mantid With Python To User Docs 2

### LINKED ISSUE #31339: Migrate Extending Mantid with Python Course to User docs
The final user-training course, [Extending Mantid With Python](https://www.mantidproject.org/Extending_Mantid_With_Python), remains on the wiki.

As a developer this is becoming hard to maintain as it can't be versioned with the code. Migrate the content to the [user docs](https://docs.mantidproject.org) alongside the other [courses](https://docs.mantidproject.org/nightly/tutorials/index.html#training) (note that the current link just goes back to the wiki version)

### PR DESCRIPTION
**Pull request by Matt-Cumber**
**Created new branch for release-next original PR #31475**

**The wiki pages links will also need to be updated after this PR is merged however I do not have permission to update the wiki. The link from the [main page](https://www.mantidproject.org/Main_Page) needs to be updated as well as [user documentation](https://www.mantidproject.org/Documentation.html)**

**Description of work.**
I've copied the course from https://www.mantidproject.org/Extending_Mantid_With_Python.html and created the equivalent in the user docs for better maintainability. I have also tried to correct the docs where I see fit.

**To test:**
Build the docs and look through the course for spelling errors etc. as well as references to MantidPlot or other things which don't make sense.
Do all the links work?
Are the images okay? (in terms of size)
Are the equations equivalent to the course (On the wiki these were images but I have written them using LaTeX in the docs) 

Fixes #31339 

*This does not require release notes* because **it is documentation maintenance**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/tutorials/extending_mantid_with_python/solutions_to_the_exercises/03_emwp_sol.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        def PyExec(self):
+            from mantid.simpleapi import Load, ApplyDiffCal, ConvertUnits, DiffractionFocussing, DeleteWorkspace
+
+            # Load file to workspace
+            _tmpws = Load(Filename=self.getPropertyValue("Filename"))
+
+            # Apply Calibration and Convert Units
+            calfile = self.getProperty("CalFilename").value
+            ApplyDiffCal(InstrumentWorkspace=_tmpws, CalibrationFile=calfile)
+            _tmpws = ConvertUnits(InputWorkspace=_tmpws, Target="dSpacing")
```
Comment (DanielMurphy22) [inline docs/source/tutorials/extending_mantid_with_python/solutions_to_the_exercises/03_emwp_sol.rst:64]: @DannyHindson can you confirm if this new solution is equivalent to the [old one](https://www.mantidproject.org/Extending_Mantid_With_Python__Exercise_3_Solution.html)??

File: docs/source/tutorials/extending_mantid_with_python/python_algorithms_and_workspaces/06_exercise_3.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* ``OutputWorkspace``: An output WorkspaceProperty to hold the final result.
+
+The steps the algorithm should perform are:
+
+#. Use the :ref:`Load <algorithms.Load.v1>` algorithm to load the TOF
+   data
+#. Apply the calibration file using
+   :ref:`ApplyDiffCal <algorithms.ApplyDiffCal.v1>`
+#. Run :ref:`ConvertUnits <algorithms.ConvertUnits.v1>` on the TOF data
+   to convert to ``dSpacing``
```
Comment (DanielMurphy22) [inline docs/source/tutorials/extending_mantid_with_python/python_algorithms_and_workspaces/06_exercise_3.rst:24]: @DannyHindson here's the related exercise instructions if you can check these make sense. The old exercise is here:  https://www.mantidproject.org/Extending_Mantid_With_Python__Exercise_3.html

**Review by DanielMurphy22 — COMMENTED** [review]
Epic! Very good job! All my requests are fixed and the course is looking really good! Also, well done on increasing the equation size! I'm ready to approve, but I'll wait to see if Danny is happy with your change to Exercise and Solution 3!

File: docs/source/tutorials/extending_mantid_with_python/solutions_to_the_exercises/03_emwp_sol.rst (hunk 1 above)
Comment (DannyHindson) [reply docs/source/tutorials/extending_mantid_with_python/solutions_to_the_exercises/03_emwp_sol.rst:64]: Yes that looks good

File: docs/source/tutorials/extending_mantid_with_python/python_algorithms_and_workspaces/06_exercise_3.rst (hunk 2 above)
Comment (DannyHindson) [reply docs/source/tutorials/extending_mantid_with_python/python_algorithms_and_workspaces/06_exercise_3.rst:24]: Where it says "using the same cal file as the previous step". It's actually the previous step but one now so strictly speaking it should say "earlier step" perhaps. But no big deal if you'd like to get this merged. Looks fine apart from that

Comment (Matt-Cumber) [reply docs/source/tutorials/extending_mantid_with_python/python_algorithms_and_workspaces/06_exercise_3.rst:24]: Good catch, I will quickly update this now

**Review by DanielMurphy22 — DISMISSED** [review]
Well done! Good to go!

**Review by martyngigg — CHANGES_REQUESTED** [review]
This is really great to have this ported over, especially now the wiki is read only.

Just a small thing to pick up on with the algorithm references.

File: docs/source/algorithms/ApplyDiffCal-v1.rst [hunk 3]
```diff
@@ -1,3 +1,5 @@
+.. _algorithms.ApplyDiffCal.v1:
```
[the remark this answers is not in the document]
Comment (Matt-Cumber) [reply docs/source/algorithms/ApplyDiffCal-v1.rst:1]: Thanks Martyn

**Review by DanielMurphy22 — APPROVED** [review]
Algorithm links work well, and good to have them linked in this way!

**Review by martyngigg — APPROVED** [review]

**Pull request merged**